### PR TITLE
[Improve][Connector-V2][IoTdb] Move iotdb connector e2e testcase into seatunnel-connector-v2-e2e

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

#3616

### CODE REVIEW

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-iotdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/iotdb/IoTDBIT.java [hunk 1]
```diff
@@ -53,7 +55,7 @@
 import java.util.stream.Stream;
 
 @Slf4j
-public class IoTDBIT extends FlinkContainer {
+public class IoTDBIT extends TestSuiteBase implements TestResource {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-iotdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/iotdb/IoTDBIT.java:62]:
```suggestion
@DisabledOnContainer(value = {}, type = {EngineType.SEATUNNEL,EngineType.SPARK}, disabledReason = xxx)
public class IoTDBIT extends TestSuiteBase implements TestResource {
```

Disable Spark Engine：
There is a conflict of thrift version between IoTDB and Spark.Therefore, you need to execute rm -f $SPARK_HOME/jars/libthrift* and cp $IOTDB_HOME/lib/libthrift* $SPARK_HOME/jars/ to resolve it.


image

Comment (Carl-Zhou-CN) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-iotdb-e2e/src/test/java/org/apache/seatunnel/e2e/connector/iotdb/IoTDBIT.java:62]: Ok, thank you

**Comment by hailin0** [comment]
Commit Message Titile：[Improve][Connector-V2][IoTdb] Move iotdb connector e2e testcase into `seatunnel-connector-v2-e2e`

**Review by hailin0 — APPROVED** [review]
LGTM

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**